lib/spack/spack/relocate.py

```diff
@@ -275,55 +275,50 @@ def modify_macho_object(cur_path, rpaths, deps, idpath,
     return
 
 
-def modify_object_machotools(cur_path, rpaths, deps, idpath,
-                             new_rpaths, new_deps, new_idpath):
+def modify_object_macholib(cur_path, old_dir, new_dir):
     """
     Modify MachO binary path_name by replacing old_dir with new_dir
     or the relative path to spack install root.
     The old install dir in LC_ID_DYLIB is replaced with the new install dir
-    using py-machotools
+    using py-macholib
     The old install dir in LC_LOAD_DYLIB is replaced with the new install dir
-    using py-machotools
+    using py-macholib
     The old install dir in LC_RPATH is replaced with the new install dir using
-    using py-machotools
+    using py-macholib
     """
     if cur_path.endswith('.o'):
         return
     try:
-        import machotools
+        from macholib.MachO import MachO
+        from macholib.mach_o import LC_RPATH
     except ImportError as e:
-        raise MissingMachotoolsException(e)
-    rewriter = machotools.rewriter_factory(cur_path)
-    if machotools.detect.is_dylib(cur_path):
-        if not new_idpath == idpath:
-            rewriter.install_name = new_idpath
-    for orig, new in zip(deps, new_deps):
-        if not orig == new:
-            rewriter.change_dependency(orig, new)
-    rewriter.commit()
-    return
+        raise MissingMacholibException(e)
 
+    def match_func(cpath):
+        return str(cpath).replace(old_dir, new_dir)
+
+    dll = MachO(cur_path)
+    dll.rewriteLoadCommands(match_func)
+    for header in dll.headers:
+        for (idx, (lc, cmd, data))  in enumerate(header.commands):
+            if lc.cmd == LC_RPATH:
+                rpath = data
+                rpath = rpath.strip('\x00')
+                new_rpath = match_func(rpath)
+                header.rewriteDataForCommand(idx, new_rpath)
 
-def machotools_get_paths(path_name):
-    """
-    Examines the output of otool -l path_name for these three fields:
-    LC_ID_DYLIB, LC_LOAD_DYLIB, LC_RPATH and parses out the rpaths,
-    dependiencies and library id.
-    Returns these values.
-    """
     try:
-        import machotools
-    except ImportError as e:
-        raise MissingMachotoolsException(e)
-    idpath = None
-    rpaths = list()
-    deps = list()
-    rewriter = machotools.rewriter_factory(path_name)
-    if machotools.detect.is_dylib(path_name):
-        idpath = rewriter.install_name
-    rpaths = rewriter.rpaths
-    deps = rewriter.dependencies
-    return rpaths, deps, idpath
+        f = open(dll.filename, 'rb+')
+        for header in dll.headers:
+            f.seek(0)
+            dll.write(f)
+        f.seek(0, 2)
+        f.flush()
+        f.close()
```

Comment:
Did you ever try this? Nothing here makes sense 😕

Reply:
It's been so long I don't recall. Feel free to remove.

Reply:
I probably cribbed it from machotools.

Reply:
I needed a pure python implementation so I could relocate mach-o binaries when installing buildcaches on linux.